indexing-service/src/main/java/io/druid/indexing/common/task/Task.java

```diff
@@ -138,4 +139,10 @@
    * @throws Exception if this task failed
    */
   public TaskStatus run(TaskToolbox toolbox) throws Exception;
+
+  public Map<String, Object> getContext();
+
+  public <ContextType> ContextType getContextValue(String key);
```

Comment:
As per prior comment. Suggest simply making Object and make the caller handle proper type checking.